pxc/jenkins/pxc56-pipeline.groovy

```diff
@@ -0,0 +1,261 @@
+                        exit 1
+                    fi
+                    rm -f ${WORKSPACE}/VERSION-${BUILD_NUMBER}
+                '''
+            }
+        }
+        stage('Check out and Build Galera/PXB') {
+            parallel {
+                stage('Build Galera library') {
+                    agent { label 'docker' }
+                    steps {
+                        git branch: 'PXC-3285-add-pipeline-jobs', url: 'https://github.com/Percona-Lab/jenkins-pipelines'
+                        echo 'Checkout Galera library'
+                        sh '''
+                            # sudo is needed for better node recovery after compilation failure
+                            # if building failed on compilation stage directory will have files owned by docker user
+                            sudo git reset --hard
+                            sudo git clean -xdf
+                            sudo rm -rf sources
+                            ./pxc/local/checkout56 GALERA3
+                        '''
+                        echo 'Build GALERA3'
+                        withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: 'c42456e5-c28d-4962-b32c-b75d161bff27', secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+                            sh '''
+                                sg docker -c "
+                                    if [ \$(docker ps -q | wc -l) -ne 0 ]; then
+                                        docker ps -q | xargs docker stop --time 1 || :
+                                    fi
+                                    ./pxc/docker/run-build-galera3 ${DOCKER_OS}
+                                " 2>&1 | tee build.log
+                             
+                                if [[ -f \$(ls pxc/sources/galera3/results/libgalera_smm.so | head -1) ]]; then
+                                    until aws s3 cp --no-progress --acl public-read pxc/sources/galera3/results/libgalera_smm.so s3://pxc-build-cache/${BUILD_TAG}/libgalera_smm.so; do
+                                        sleep 5
+                                    done
+                                    until aws s3 cp --no-progress --acl public-read pxc/sources/galera3/results/garbd s3://pxc-build-cache/${BUILD_TAG}/garbd; do
+                                        sleep 5
+                                    done
+                                else
+                                    echo cannot find compiled archive
+                                    exit 1
+                                fi
+                            '''
+                        }
+                    }
+                }
+                stage('Build PXB23') {
+                    agent { label 'docker' }
+                    steps {
+                        git branch: 'PXC-3285-add-pipeline-jobs', url: 'https://github.com/Percona-Lab/jenkins-pipelines'
```

Comment:
please use 'master' branch  for https://github.com/Percona-Lab/jenkins-pipeline

Reply:
done